### PR TITLE
Feature/2064 assignment without session

### PR DESCRIPTION
**Pull request by timohuber**

### CODE REVIEW

File: pool/web/handler/admin_session.ml [hunk 1]
```diff
@@ -106,14 +121,19 @@ let new_helper req page =
              flash_fetcher
            |> Lwt_result.ok
          | `New ->
```
Comment (mabiede) [inline pool/web/handler/admin_session.ml:123]: How about adding the the next match statement directly in the parent one as "when ..."

```suggestion
         | `New when CCOption.is_some experiment.Experiment.online_study ->
           ...
         | `New ->
```

File: pool/web/view/page/page_admin_session.ml [hunk 2]
```diff
@@ -511,6 +502,44 @@ let reschedule_session
          experiment)
 ;;
 
+(* 
+
+   let data_table ({ Pool_context.language; _ } as context) experiment
```
Comment (mabiede) [inline pool/web/view/page/page_admin_session.ml:507]: delete

File: pool/app/experiment/entity.ml [hunk 3]
```diff
@@ -79,6 +79,48 @@ module ShowExternalDataIdLinks = struct
...
+    let open CCResult.Infix in
+    let open Pool_common.Message in
+    let open Uri in
+    let invalid_error = Error (Invalid Field.SurveyUrl) in
+    let trimmed = CCString.trim str in
+    let uri = of_string trimmed in
+    try
+      let* () =
+        match scheme uri with
+        | Some ("http" | "https") -> Ok ()
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/experiment/entity.ml:101]: IMO, this is the responsibility of the survey software. Most URLs are redirected to https anyways.

File: pool/app/experiment/entity.ml [hunk 4]
```diff
@@ -79,6 +79,48 @@ module ShowExternalDataIdLinks = struct
...
+      let* (_ : string * string list) =
+        let value = Format.asprintf "{%s}" Field.(show CallbackUrl) in
+        uri
+        |> query
+        |> CCList.find_opt (fun (_, values) ->
+          values
+          |> CCList.head_opt
+          |> CCOption.map_or ~default:false (CCString.equal value))
+        |> CCOption.to_result (FieldRequired Field.CallbackUrl)
+      in
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/experiment/entity.ml:113]: Yes, I am checking if one of the values is the callback URL. The naming is irrelevant IMO.

File: pool/app/time_window/repo/repo.ml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    find_request
+    (Session.Id.value id)
+  ||> CCOption.to_result Pool_common.Message.(NotFound Field.Session)
+;;
+
+let find_overlapping_request exclude =
+  let end_at =
+    "DATE_ADD(pool_sessions.start, INTERVAL pool_sessions.duration SECOND)"
+  in
+  (* TODO: How to deal with start or and at the same time? *)
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/time_window/repo/repo.ml:73]: I did not change it, so time windows cannot start at the time another one ends. To achieve that, the end of the existing time window can simply be edited

File: pool/app/experiment/experiment.mli [hunk 6]
```diff
@@ -65,6 +65,48 @@ module ShowExternalDataIdLinks : sig
...
 
+module AssignmentWithoutSession : sig
+  include Pool_common.Model.BooleanSig
+end
+
+module SurveyUrl : sig
+  include Pool_common.Model.StringSig
+end
+
+module OnlineStudy : sig
```
[the remark this answers is not in the document]
Comment (timohuber) [reply pool/app/experiment/experiment.mli:76]: OnlineExperiment it is.

File: pool/web/handler/admin_session.ml (hunk 1 above)
Comment (timohuber) [reply pool/web/handler/admin_session.ml:123]: It seems that does not work with polymorphic variants as it does with variant types. The compiler wants me to handle the case of `| New -> ` which does not exist.

Comment (mabiede) [reply pool/web/handler/admin_session.ml:123]: Do we have this separation multiple times? should a ` type handleCreateAs = New | FollowUp` be defined?

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/web/handler/admin_session.ml (hunk 1 above)
Comment (timohuber) [reply pool/web/handler/admin_session.ml:123]: There are only 2 pattern matches.

**Review by mabiede — APPROVED** [review]

**Pull request merged**